controllers/nnf_node_ec_data_controller.go

```diff
@@ -0,0 +1,205 @@
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the NnfNodeECData object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *NnfNodeECDataReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	// your logic here
+
+	return ctrl.Result{}, nil
+}
+
+func (r *NnfNodeECDataReconciler) NewPersistentStorageInterface(name string, readOnly bool) (persistent.PersistentStorageApi, error) {
+	return &crdPersistentStorageInterface{reconciler: r, name: name}, nil
+}
+
+// Implementation of Persistent Storage API for CRD Persistent Storage
+type crdPersistentStorageInterface struct {
+	reconciler *NnfNodeECDataReconciler
+	name       string
+}
+
+func (*crdPersistentStorageInterface) Close() error {
+	return nil
+}
+
+func (psi *crdPersistentStorageInterface) View(fn func(persistent.PersistentStorageTransactionApi) error) error {
+	data := nnfv1alpha1.NnfNodeECData{}
+	if err := psi.reconciler.Get(context.TODO(), psi.reconciler.NamespacedName, &data); err != nil {
+		return err
+	}
+
+	return fn(persistent.NewBase64PersistentStorageTransaction(data.Status.Data[psi.name]))
+}
+
+func (psi *crdPersistentStorageInterface) Update(fn func(persistent.PersistentStorageTransactionApi) error) error {
+
+Retry:
+
+	data := nnfv1alpha1.NnfNodeECData{}
+	if err := psi.reconciler.Get(context.TODO(), psi.reconciler.NamespacedName, &data); err != nil {
+		return err
+	}
+
+	if data.Status.Data == nil {
+		data.Status.Data = make(map[string]nnfv1alpha1.NnfNodeECPrivateData)
+	}
+
+	if _, found := data.Status.Data[psi.name]; !found {
+		data.Status.Data[psi.name] = make(nnfv1alpha1.NnfNodeECPrivateData)
+	}
+
+	if err := fn(persistent.NewBase64PersistentStorageTransaction(data.Status.Data[psi.name])); err != nil {
+		return err
+	}
+
+	if err := psi.reconciler.Status().Update(context.TODO(), &data); err != nil {
+		if errors.IsConflict(err) {
+			goto Retry
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (psi *crdPersistentStorageInterface) Delete(key string) error {
+
+Retry:
+
+	data := nnfv1alpha1.NnfNodeECData{}
+	if err := psi.reconciler.Get(context.TODO(), psi.reconciler.NamespacedName, &data); err != nil {
+		return err
+	}
+
+	delete(data.Status.Data[psi.name], key)
+
+	if err := psi.reconciler.Status().Update(context.TODO(), &data); err != nil {
+		if errors.IsConflict(err) {
+			goto Retry
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *NnfNodeECDataReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.Add(r); err != nil {
```

Comment:
Add a comment like this one that you have in nnf_node_controller:
```// Causes NnfNodeReconciler.Start() to be called.```
Sometimes these clues help to stitch things together;  in this case it takes a bit to work it out even with the help of VSCode.